### PR TITLE
add logic for detect deleted page and update only updated page

### PR DESCRIPTION
**Pull request by Kadi-7**

### CODE REVIEW

**Comment by emuguy1** [comment]
Looks great olease resolve the conflict with main branch so we can merge this

**Comment by zenzeii** [comment]
Nice looks good! Thank you for moving some functionality to an extra function 👍

File: QAChat/Data_Processing/confluence_preprocessor.py [hunk 1]
```diff
@@ -225,15 +215,62 @@ def add_content_of_pdf_to_all_page_information(self, page_id) -> str:
...
+                self.remove_from_db(i)  # remove it from DB
+
+    def remove_from_db(self, id):
+        # loop over max number in chunk id and remove all the rows from DB
+        for i in range(0, int(self.chunk_id_lookup_table[id]) + 1):
+            self.supabase_client.table("data_embedding").delete().eq("metadata->>id", str(id) + "_" + str(i)).execute()
+
 
 if __name__ == "__main__":
     cp = ConfluencePreprocessor()
```
Comment (amela16) [inline QAChat/Data_Processing/confluence_preprocessor.py:276]: Do I have to do something before running the main? Like adding a URL somewhere? When I run the main I get this error:
Traceback (most recent call last):
  File "/home/zitrone/Desktop/Amos/amos2023ss03-qachat/QAChat/Data_Processing/confluence_preprocessor.py", line 276, in <module>
    cp = ConfluencePreprocessor()
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zitrone/Desktop/Amos/amos2023ss03-qachat/QAChat/Data_Processing/confluence_preprocessor.py", line 35, in __init__
    self.confluence = Confluence(
                      ^^^^^^^^^^^
  File "/home/zitrone/Desktop/Amos/amos2023ss03-qachat/venv/lib/python3.11/site-packages/atlassian/confluence.py", line 30, in __init__
    if ("atlassian.net" in url or "jira.com" in url) and ("/wiki" not in url):
        ^^^^^^^^^^^^^^^^^^^^^^
TypeError: argument of type 'NoneType' is not iterable

Process finished with exit code 1

Comment (Kadi-7) [reply QAChat/Data_Processing/confluence_preprocessor.py:276]: please use the new version tokens.env. Seems like the url you use URL without wiki

**Pull request merged**